com.unity.shadergraph/Editor/GraphUI/ShaderGraphEditorWindow.cs

```diff
@@ -69,18 +69,14 @@ protected override BlankPage CreateBlankPage()
 
         protected override bool CanHandleAssetType(IGraphAssetModel asset)
         {
-            // TODO (Sai):
-            // Figure out a fitting place to initialize these
-            m_PreviewManager = new PreviewManager(asset.GraphModel as ShaderGraphModel);
-            m_GraphViewStateObserver = new GraphViewStateObserver(GraphView.GraphViewState, m_PreviewManager);
-            GraphView.GraphTool.ObserverManager.RegisterObserver(m_GraphViewStateObserver);
-
-            // Set reference so its available during command handling
-            m_GraphTool.previewManager = m_PreviewManager;
+            return asset is ShaderGraphAssetModel;
+        }
 
-            ShaderGraphState.RegisterCommandHandlers(GraphTool, GraphView, asset.GraphModel as ShaderGraphModel, GraphTool.Dispatcher);
+        protected void OnBecameVisible()
+        {
+            m_PreviewManager.Initialize(GraphTool.ToolState.GraphModel as ShaderGraphModel);
```

Comment:
GraphModel here doesn't seem guaranteed to be initialized yet.

Reply:
thanks for finding this, didnt run into it when just opening an existing asset from disk, which makes sense, guess I just need to check if `GraphModel` is null to see if we're actually opening a graph asset